Source/Core/DolphinQt2/QtUtils/SignalDaemon.h

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QObject>
+
+class QSocketNotifier;
+
+// Loosely based on https://doc.qt.io/qt-5.9/unix-signals.html
+class SignalDaemon : public QObject
+{
+  Q_OBJECT
+
+public:
+  SignalDaemon(QObject* parent);
```

Comment:
Could you add a destructor to `close(2)` s_sigterm_fd[0] and s_sigterm_fd[1]? Not that it really matters, but I feel like that would be cleaner than leaking sockets and relying on the kernel to clean up for us.